lib/internal/test_runner/runner.js

```diff
@@ -88,10 +88,12 @@ const kCanceledTests = new SafeSet()
 
 let kResistStopPropagation;
 
-function createTestFileList() {
+function createTestFileList(patterns) {
   const cwd = process.cwd();
-  const hasUserSuppliedPattern = process.argv.length > 1;
-  const patterns = hasUserSuppliedPattern ? ArrayPrototypeSlice(process.argv, 1) : [kDefaultPattern];
+  const hasUserSuppliedPattern = patterns != null;
+  if (!patterns || patterns.length === 0) {
+    patterns = [kDefaultPattern];
```

Comment:
I would assign the default patterns in the `run()` validation. You can do that in this PR, otherwise, I will probably do it in a follow up.